projectq/backends/_sim/_pysim.py

```diff
@@ -239,22 +239,60 @@ def get_expectation_value(self, terms_dict, ids):
         Returns:
             Expectation value
         """
-        X = [[0., 1.], [1., 0.]]
-        Y = [[0., -1j], [1j, 0.]]
-        Z = [[1., 0.], [0., -1.]]
-        gates = [X, Y, Z]
         expectation = 0.
         current_state = _np.copy(self._state)
         for (term, coefficient) in terms_dict:
-            for local_op in term:
-                qb_id = ids[local_op[0]]
-                self.apply_controlled_gate(gates[ord(local_op[1]) - ord('X')],
-                                           [qb_id], [])
+            self._apply_term(term, ids)
             delta = coefficient * _np.vdot(current_state, self._state).real
             expectation += delta
             self._state = _np.copy(current_state)
         return expectation
 
+    def emulate_time_evolution(self, terms_dict, time, ids, ctrlids):
+        """
+        Applies exp(-i*time*H) to the wave function, i.e., evolves under
+        the Hamiltonian H for a given time.
```

Comment:
Maybe add that this function is general and allows the Hamiltonian to be a sum of non commuting terms